### PR TITLE
Convert schemaform address to use updateSchema

### PR DESCRIPTION
**Pull request by jbalboni**
Connects to https://github.com/department-of-veterans-affairs/vets.gov-team/issues/2691

This updates our address definition to use the `updateSchema` option instead of a whole custom component. This should cut down on bundle size and make things more consistent.

Should be no visual changes.

### CODE REVIEW

**Review by cehsu — APPROVED** [review]

File: src/js/common/schemaform/definitions/address.js [hunk 1]
```diff
@@ -30,14 +48,69 @@ export function schema(currentSchema, isRequired = false) {
...
+        if (country === 'USA') {
+          stateList = usaStates;
+        } else if (country === 'CAN') {
+          stateList = canProvinces;
+        } else if (country === 'MEX') {
+          stateList = mexStates;
+        }
+
+        if (stateList) {
+          // We have a list and it's different, so we need to make schema updates
```
Comment (cvalarida) [inline src/js/common/schemaform/definitions/address.js]: Could this / should this be done in a selector? As long as we're already using reselect... :man_shrugging:

**Review by cvalarida — APPROVED** [review]

File: src/js/common/schemaform/definitions/address.js [hunk 2]
```diff
@@ -30,14 +48,69 @@ export function schema(currentSchema, isRequired = false) {
...
+export function uiSchema(label = 'Address', useStreet3 = false) {
+  let fieldOrder = ['country', 'street', 'street2', 'street3', 'city', 'state', 'postalCode'];
+  if (!useStreet3) {
+    fieldOrder = fieldOrder.filter(field => field !== 'street3');
+  }
+
   return {
     'ui:title': label,
-    'ui:field': 'address',
+    // 'ui:field': 'address',
```
Comment (cehsu) [inline src/js/common/schemaform/definitions/address.js]: @jbalboni could you please explain the comment here?

File: src/js/common/schemaform/definitions/address.js (hunk 1 above)
Comment (jbalboni) [inline src/js/common/schemaform/definitions/address.js]: Ha, good point, I completely forgot about reselect. I will give that a shot.

File: src/js/common/schemaform/definitions/address.js (hunk 2 above)
Comment (jbalboni) [inline src/js/common/schemaform/definitions/address.js]: I just need to remove it entirely. I commented out when I first started this work.

**Review by cvalarida — APPROVED** [review]
Excellent! Yay, selectors!

**Pull request merged**